application/core/LSYii_Application.php

```diff
@@ -591,4 +591,163 @@ private function setSessionByDB($aApplicationConfig)
+            }
+
+            foreach ($additionalFields as $additionalField) {
+                if (isset($archivedResponse->{$additionalField}) && isset($targetSchema->columns[$additionalField])) {
+                    $dataRow[$additionalField] = $archivedResponse->{$additionalField};
+                }
+            }
+
+            $beforeDataEntryImport = new PluginEvent('beforeDataEntryImport');
+            $beforeDataEntryImport->set('iSurveyID', $surveyId);
+            $beforeDataEntryImport->set('oModel', $targetResponse);
+            App()->getPluginManager()->dispatchEvent($beforeDataEntryImport);
+
+            if ($targetResponse->validate()){
+                $batchData[] = $dataRow;
+            }
+            if (count($batchData) % 500 === 0) {
+                if ($preserveIDs) {
+                    switchMSSQLIdentityInsert("survey_$surveyId", true);
+                }
+                $builder = App()->db->getCommandBuilder();
+                $command = $builder->createMultipleInsertCommand($tableName, $batchData);
+                $importedResponses += $command->execute();
+                if ($preserveIDs) {
+                    switchMSSQLIdentityInsert("survey_$surveyId", false);
+                }
+                $batchData = [];
+            }
+
+            unset($targetResponse);
+        }
+
+        if (count($batchData)) {
+            if ($preserveIDs) {
+                switchMSSQLIdentityInsert("survey_$surveyId", true);
+            }
+            $builder = App()->db->getCommandBuilder();
+            $command = $builder->createMultipleInsertCommand($tableName, $batchData);
+            $importedResponses += $command->execute();
+            if ($preserveIDs) {
+                switchMSSQLIdentityInsert("survey_$surveyId", false);
+            }
+        }
+        return $importedResponses;
+    }
+
+    /**
+     * Copying all data from source table to a target table having the same structure
+     *
+     * @param string $source
+     * @param string $destination
+     * @return integer number of rows affected by the execution.
+     * @throws CDbException execution failed
+     */
+    public function copyFromOneTableToTheOther($source, $destination)
+    {
+        $customFilter = [
+            'mysql' => 'a.TABLE_SCHEMA = b.TABLE_SCHEMA and a.TABLE_SCHEMA = DATABASE()',
+            'mysqli' => 'a.TABLE_SCHEMA = b.TABLE_SCHEMA and a.TABLE_SCHEMA = DATABASE()',
+            'pgsql' => 'a.TABLE_CATALOG = b.TABLE_CATALOG and a.TABLE_CATALOG = current_database()',
+            'mssql' => 'a.TABLE_CATALOG = b.TABLE_CATALOG and a.TABLE_CATALOG = db_name()',
+            'sqlsrv' => 'a.TABLE_CATALOG = b.TABLE_CATALOG and a.TABLE_CATALOG = db_name()',
+        ];
+        $filter = ($customFilter[Yii::app()->db->getDriverName()] ?? '');
+        $command = "
+            select a.COLUMN_NAME as cname
+            from information_schema.columns a
```

Comment:
Afaik there is no such information_schema in other databases than mysql.
As previosly discussed I'd strongly recommend to use the schema information functions provided by Yii, instead.

Reply:
@c-schmitz this was successfully tested in PostgreSQL and SQL Server.

See https://www.postgresql.org/docs/current/information-schema.html

and

https://learn.microsoft.com/en-us/sql/relational-databases/system-information-schema-views/system-information-schema-views-transact-sql?view=sql-server-ver16

Can you provide the details of the tests that failed for you?